backend/matching-service/src/controllers/matching.controller.ts

```diff
@@ -1,42 +1,26 @@
 import { ITypedBodyRequest } from '@repo/request-types'
 import { Response } from 'express'
-import { ValidationError } from 'class-validator'
-import { UserQueueRequestDto } from '../types/UserQueueRequestDto'
+import { UserQueueRequest, UserQueueRequestDto } from '../types/UserQueueRequestDto'
 import mqConnection from '../services/rabbitmq.service'
-import { IUserQueueMessage } from '../types/IUserQueueMessage'
+import { randomUUID } from 'crypto'
+import { WebSocketMessageType } from '@repo/ws-types'
+import wsConnection from '../server'
 
-export async function addUserToMatchingQueue(
-    request: ITypedBodyRequest<UserQueueRequestDto>,
-    response: Response
-): Promise<void> {
-    const createDto = UserQueueRequestDto.fromRequest(request)
-    const errors = await createDto.validate()
-    if (errors.length) {
-        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
-        response.status(400).json(errorMessages).send()
-        return
-    }
-
-    // Add logic to generate and return 2 WS IDs
-
-    const message: IUserQueueMessage = { ...createDto, websocketId: 'testing' }
-    await mqConnection.sendToEntryQueue(message)
-    response.status(200).send()
+export async function generateWS(request: ITypedBodyRequest<void>, response: Response): Promise<void> {
+    const websocketID = randomUUID()
+    response.status(200).send({ websocketID: websocketID })
 }
 
-// This will change after the WS is implemented
-export async function removeUserFromMatchingQueue(
-    request: ITypedBodyRequest<UserQueueRequestDto>,
-    response: Response
-): Promise<void> {
-    const createDto = UserQueueRequestDto.fromRequest(request)
+export async function addUserToMatchmaking(data: UserQueueRequest): Promise<void> {
+    const createDto = UserQueueRequestDto.fromJSON(data)
     const errors = await createDto.validate()
     if (errors.length) {
-        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
-        response.status(400).json(errorMessages).send()
         return
     }
+    await mqConnection.sendToEntryQueue(createDto)
```

Comment:
the user is added to the queue before there is any guarantee that the WS connection has been established, will probably have to tie this to the onconnect event in WS .

Reply:
this controller is only called on message from websocket under the handlemessage method and so there is alr a connection established

Reply:
icic okay